File: acwing/算法提高课/第一章动态规划/1020.潜水员/1020.潜水员.py
M = 25
N = 85

f = [[float("inf") for _ in range(N)] for _ in range(N)]
f[0][0] = 0

m, n = map(int, input().split())

k = int(input())

for _ in range(k):
    a, b, c = map(int, input().split())
    for i in range(m, -1, -1):
        for j in range(n, -1, -1):
            f[i][j] = min(f[i][j], f[max(0, i - a)][max(0, j - b)] + c)

print(f[m][n])

File: acwing/算法提高课/第一章动态规划/1020.潜水员/1020.潜水员.cpp
#include <iostream>
#include <algorithm>
#include <cstring>

using namespace std;

const int M = 25, N = 85;
int f[M][N];

int main()
{
    int m, n;
    cin >> m >> n;
    int k;
    cin >> k;
    memset(f, 0x3f, sizeof(f));
    f[0][0] = 0;
    while (k--)
    {
        int a, b, c;
        cin >> a >> b >> c;
        for (int i = m; i >= 0; i--)
        {
            for (int j = n; j >= 0; j--)
            {
                f[i][j] = min(f[i][j], f[max(0, i - a)][max(0, j - b)] + c);
            }
        }
    }
    cout << f[m][n] << endl;

    return 0;
}